clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -4832,8 +4832,7 @@ static void handleSYCLDeviceIndirectlyCallableAttr(Sema &S, Decl *D,
 static void handleSYCLRegisterNumAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   auto *VD = cast<VarDecl>(D);
   if (!VD->hasGlobalStorage()) {
-    S.Diag(AL.getLoc(), diag::err_sycl_attibute_cannot_be_applied_here)
-        << AL << 0;
+    S.Diag(AL.getLoc(), diag::err_sycl_attribute_internal_function) << AL << 0;
```

Comment:
Actually, I just took a closer look. Why is a diagnostic about static function/function in anonymous namespace being generated for a variable. This is out of scope of your patch but this looks like incorrect diagnostic?

Reply:
Looks like it was added by @kbobrovs in scope of https://github.com/intel/llvm/pull/1756 . Usage of this diagnostic seems a bit odd and untested.

Reply:
Do I have to fix it?

Reply:
If we'd like to upstream at some point, I'd recommend fixing it. (I think it's in scope for this patch given that the diagnostic quality is being regressed, FWIW).

Reply:
Ok. `register_num` has only 'GlobalVar' in subjects list, so it seems the part of the code where this diagnostic was incorrectly used is dead code. Removed.

Reply:
> If we'd like to upstream at some point

Go for it! :-)